Client/src/main/java/com/minecrafttas/tasbattle/Tickratechanger.java

```diff
@@ -14,17 +14,35 @@ public class Tickratechanger {
 	@Getter
 	private long msPerTick;
 	
+	private long systemTimeSinceTC = System.currentTimeMillis(); // system time passed since last tickrate change
+	private long gameTime = System.currentTimeMillis(); // game time passed (since last tickrate update)
+	
 	public void changeTickrate(float tickrate) {
 		if(tickrate<=0 || tickrate > 1000) return;
 		TASBattle.LOGGER.debug("Changing tickrate to {}", tickrate);
 		
+		long millis = System.currentTimeMillis();
+
+		// calculate time passed without tickrate changing
+		long timePassed = millis - this.systemTimeSinceTC;
+		this.gameTime += (long) (timePassed * this.gamespeed);
+		this.systemTimeSinceTC = millis;
+		
 		this.tickrate=tickrate;
 		this.gamespeed = tickrate/20;
 		this.msPerTick = (long) (1000L/tickrate);
 		
 		changeClientTickrate(msPerTick);
 	}
```

Comment:
i get that server tickrate changer is useful, but they don't need another method just to set the client tickrate